inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -178,13 +179,27 @@ ie::CNNNetwork FrontEnd::convertNetwork(ie::CNNNetwork& network) {
         ngraph::opset5::Convolution::type_info,
         ngraph::opset5::GroupConvolution::type_info
     });
-
+    const std::vector<std::pair<ngraph::element::Type, ngraph::element::Type>> convert_precision_list {
+        {ngraph::element::i64, ngraph::element::i32},
+        {ngraph::element::u64, ngraph::element::i32},
+        {ngraph::element::u32, ngraph::element::i32},
+        {ngraph::element::u16, ngraph::element::i32},
+        {ngraph::element::boolean, ngraph::element::i32},
+    };
+    for (const auto &precision : convert_precision_list) {
+        manager.register_pass<ngraph::pass::ConvertPrecision>(precision.first, precision.second, myriad_type_to_fuse);
+    }
     manager.register_pass<vpu::DynamicToStaticShape>();
     manager.register_pass<vpu::EliminateShapeOfAfterDSR>();
     manager.register_pass<vpu::ConvertExtractImagePatchesToReorgYolo>();
     manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
     manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
+
+    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::i64, ngraph::element::i32, myriad_type_to_fuse);
     manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
+    for (const auto &precision : convert_precision_list) {
+        manager.register_pass<ngraph::pass::ConvertPrecision>(precision.first, precision.second, myriad_type_to_fuse);
```

Comment:
Do we need to run all of them?

Reply:
I'm not sure about all of them, but when I worked on this task I saw what some nodes have `bool` precision after conversion.
We try to protect ourselves from unexpected precisions.